library/src/main/java/org/apache/apex/malhar/lib/wal/FSWindowDataManager.java

```diff
@@ -416,7 +418,11 @@ public void save(Object object, long windowId) throws IOException
 
     byte[] windowIdBytes = Longs.toByteArray(windowId);
     writer.append(new Slice(windowIdBytes));
+
+    //this call will copy all data to the file input stream, so the data is useless any more after call return
```

Comment:
The stream memory was reset all instead of reset to certain window, so beginWindow(), endWindow() don't need to be called